src/net.cpp

```diff
@@ -193,30 +193,30 @@ bool IsPeerAddrLocalGood(CNode *pnode)
            IsReachable(addrLocal.GetNetwork());
 }
 
-// pushes our own address to a peer
 void AdvertiseLocal(CNode *pnode)
```

Comment:
Since you're completely refactoring this function and both call sites, you may as well fix the signature to take a `CNode&` instead of a `CNode*`. I think you may also be able to make it a const reference.

Reply:
I don't think const reference is possible due to `PushAddress` call.